pkg/manager/impl/execution_manager.go

```diff
@@ -235,26 +235,46 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask,
 	taskResourceLimits := runtimeInterfaces.TaskResourceSet{}
 
 	// For resource values, we prefer to use the limits set in the application config over the set resource values.
-	if len(configResourceLimits.CPU) > 0 {
-		cpuLimit = configResourceLimits.CPU
-	} else if cpuIndex >= 0 {
-		cpuLimit = resourceRequestEntries[cpuIndex].Value
-	}
-	taskResourceLimits.CPU = cpuLimit
-	if len(configResourceLimits.Memory) > 0 {
-		memoryLimit = configResourceLimits.Memory
-	} else if memoryIndex >= 0 {
-		memoryLimit = resourceRequestEntries[memoryIndex].Value
-	}
-	taskResourceLimits.Memory = memoryLimit
-	if len(taskResourceLimits.GPU) == 0 && len(configResourceLimits.GPU) > 0 {
+	if !configResourceLimits.CPU.IsZero() {
+		taskResourceLimits.CPU = configResourceLimits.CPU
+	} else {
+		if cpuIndex >= 0 {
+			cpuLimit = resourceRequestEntries[cpuIndex].Value
+		}
+		cpu, err := resource.ParseQuantity(cpuLimit)
+		if err != nil {
+			logger.Infof(ctx, "Failed to parse user cpu limit from task spec [%s] with err [%+v]", cpuLimit, err)
```

Comment:
Shouldn't this fail the task?

Reply:
I guess this should have failed the compilation...

Reply:
Maybe make it `Errorf` and indicate this should have failed in the compiler earlier?

Reply:
oh yeah this is why i filed https://github.com/flyteorg/flyte/issues/1403 to clean this up so we can easily enforce this at serialization/compilation

Reply:
upgraded to an error